From ba2543cae33e8a1db3045e426ae2082384993112 Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Fri, 22 Nov 2019 17:25:58 +0900
Subject: [PATCH 14/66] Add and show gooroom protocol logo with short brand 
 text and fake scheme gooroom:// instead of chrome:// in omnibox.

---
 chrome/app/chromium_strings.grd               | 14 ++++++
 .../chrome_location_bar_model_delegate.cc     |  6 +++
 .../views/location_bar/location_icon_view.cc  | 11 +++++
 .../views/page_info/page_info_bubble_view.cc  |  9 +++-
 components/components_chromium_strings.grd    |  2 +-
 components/omnibox/browser/BUILD.gn           |  3 ++
 .../browser/vector_icons/gooroom_product.icon | 47 +++++++++++++++++++
 .../vector_icons/gooroom_product_1x.icon      | 47 +++++++++++++++++++
 .../components_chromium_strings_ko.xtb        |  4 +-
 9 files changed, 139 insertions(+), 4 deletions(-)
 create mode 100644 components/omnibox/browser/vector_icons/gooroom_product.icon
 create mode 100644 components/omnibox/browser/vector_icons/gooroom_product_1x.icon

diff --git a/chrome/app/chromium_strings.grd b/chrome/app/chromium_strings.grd
index 7b8a2fd6b1e6..3a5ac42607c1 100644
--- a/chrome/app/chromium_strings.grd
+++ b/chrome/app/chromium_strings.grd
@@ -126,12 +126,26 @@ If you update this file, be sure also to update google_chrome_strings.grd. -->
         <part file="settings_chromium_strings.grdp" />
       </if>
 
+      <message name="IDS_PROFILES_DISCONNECT_MANAGED_PROFILE_TEXT" desc="Message explaining to the user what will happen if they disconnect the managed profile.">
+        Disconnecting <ph name="USERNAME">$1<ex>someone@example.com</ex></ph> will clear your history, bookmarks, settings, and other Chromium data stored on this device. Data stored in your Google Account will not be cleared and can be managed on <ph name="GOOGLE_DASHBOARD_LINK">&lt;a target="_blank" href="$2"&gt;</ph>Google Dashboard<ph name="END_GOOGLE_DASHBOARD_LINK">&lt;/a&gt;</ph>.
+      </message>
+      <if expr="_gooroom">
+        <!-- String "Gooroom" immediately after Gooroom protocol logo -->
+        <message name="IDS_PRODUCT_NAME" desc="The Gooroom Browser application name">
+          Gooroom Browser
+        </message>
+        <message name="IDS_SHORT_PRODUCT_NAME" desc="The Gooroom Browser application short name.">
+          Gooroom
+        </message>
+      </if>
+      <if expr="not _gooroom">
       <message name="IDS_PRODUCT_NAME" desc="The Chrome application name">
         Chromium
       </message>
       <message name="IDS_SHORT_PRODUCT_NAME" desc="The Chrome application short name.">
         Chromium
       </message>
+      </if>
       <if expr="is_win">
         <message name="IDS_SXS_SHORTCUT_NAME" desc="Unused in Chromium builds" translateable="false">
         </message>
diff --git a/chrome/browser/ui/toolbar/chrome_location_bar_model_delegate.cc b/chrome/browser/ui/toolbar/chrome_location_bar_model_delegate.cc
index dff64d7f7cba..2ddee6908981 100644
--- a/chrome/browser/ui/toolbar/chrome_location_bar_model_delegate.cc
+++ b/chrome/browser/ui/toolbar/chrome_location_bar_model_delegate.cc
@@ -179,6 +179,12 @@ const gfx::VectorIcon* ChromeLocationBarModelDelegate::GetVectorIconOverride()
     return &omnibox::kExtensionAppIcon;
 #endif
 
+#if defined(GOOROOM_BUILD)
+  // Show gooroom protocol logo prior to url "chrome://*".
+  if (url.SchemeIs(content::kGooroomUIScheme))
+    return &omnibox::kGooroomProductIcon;
+#endif // GOOROOM_BUILD
+
   return nullptr;
 }
 
diff --git a/chrome/browser/ui/views/location_bar/location_icon_view.cc b/chrome/browser/ui/views/location_bar/location_icon_view.cc
index 222fab010d58..e7abac2b3dd9 100644
--- a/chrome/browser/ui/views/location_bar/location_icon_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_icon_view.cc
@@ -125,6 +125,10 @@ bool LocationIconView::ShouldShowText() const {
   const GURL& url = location_bar_model->GetURL();
   if (url.SchemeIs(content::kChromeUIScheme) ||
       url.SchemeIs(extensions::kExtensionScheme) ||
+#if defined(GOOROOM_BUILD)
+      // Show a string "Gooroom" immediately after Gooroom protocol logo
+        url.SchemeIs(content::kGooroomUIScheme) ||
+#endif
       url.SchemeIs(url::kFileScheme)) {
     return true;
   }
@@ -140,9 +144,16 @@ base::string16 LocationIconView::GetText() const {
   if (delegate_->IsEditingOrEmpty())
     return base::string16();
 
+#if defined(GOOROOM_BUILD)
+  // Show a string "안전한 Gooroom 페이지를..." after clicking Gooroom protocol logo
+  if (delegate_->GetLocationBarModel()->GetURL().SchemeIs(content::kGooroomUIScheme)) {
+    return l10n_util::GetStringUTF16(IDS_SHORT_PRODUCT_NAME);
+  }
+#else
   if (delegate_->GetLocationBarModel()->GetURL().SchemeIs(
           content::kChromeUIScheme))
     return l10n_util::GetStringUTF16(IDS_SHORT_PRODUCT_NAME);
+#endif
 
   if (delegate_->GetLocationBarModel()->GetURL().SchemeIs(url::kFileScheme))
     return l10n_util::GetStringUTF16(IDS_OMNIBOX_FILE);
diff --git a/chrome/browser/ui/views/page_info/page_info_bubble_view.cc b/chrome/browser/ui/views/page_info/page_info_bubble_view.cc
index d6ef4c161e89..db50226c969e 100644
--- a/chrome/browser/ui/views/page_info/page_info_bubble_view.cc
+++ b/chrome/browser/ui/views/page_info/page_info_bubble_view.cc
@@ -375,7 +375,11 @@ InternalPageInfoBubbleView::InternalPageInfoBubbleView(
     text = IDS_PAGE_INFO_FILE_PAGE;
   } else if (url.SchemeIs(content::kChromeDevToolsScheme)) {
     text = IDS_PAGE_INFO_DEVTOOLS_PAGE;
-  } else if (!url.SchemeIs(content::kChromeUIScheme)) {
+  } else if (!url.SchemeIs(content::kChromeUIScheme) 
+#if defined(GOOROOM_BUILD)
+             && !url.SchemeIs(content::kGooroomUIScheme)
+#endif
+  ) {
     NOTREACHED();
   }
 
@@ -421,6 +425,9 @@ views::BubbleDialogDelegateView* PageInfoBubbleView::CreatePageInfoBubble(
   gfx::NativeView parent_view = platform_util::GetViewForWindow(parent_window);
 
   if (url.SchemeIs(content::kChromeUIScheme) ||
+#if defined(GOOROOM_BUILD)
+      url.SchemeIs(content::kGooroomUIScheme) ||
+#endif
       url.SchemeIs(content::kChromeDevToolsScheme) ||
       url.SchemeIs(extensions::kExtensionScheme) ||
       url.SchemeIs(content::kViewSourceScheme) ||
diff --git a/components/components_chromium_strings.grd b/components/components_chromium_strings.grd
index 176583712d04..26a364c8c95e 100644
--- a/components/components_chromium_strings.grd
+++ b/components/components_chromium_strings.grd
@@ -222,7 +222,7 @@
 
       <!-- Page Info -->
       <message name="IDS_PAGE_INFO_INTERNAL_PAGE" desc="Message to display in the page info bubble when the page you are on is a chrome:// page or about:something.">
-        You're viewing a secure Chromium page
+        You're viewing a secure Gooroom page
       </message>
       <message name="IDS_PAGE_INFO_SECURITY_TAB_SECURE_IDENTITY_VERIFIED" desc="The text of the identity section when the page is secure and uses a valid certificate">
         Chromium verified that <ph name="ISSUER">$1<ex>VeriSign</ex></ph> issued this website's certificate.
diff --git a/components/omnibox/browser/BUILD.gn b/components/omnibox/browser/BUILD.gn
index 8992ebf6a626..b771102f2bf7 100644
--- a/components/omnibox/browser/BUILD.gn
+++ b/components/omnibox/browser/BUILD.gn
@@ -60,6 +60,9 @@ aggregate_vector_icons("omnibox_vector_icons") {
     "star.icon",
     "switch.icon",
     "tab.icon",
+    # convert Gooroom protocol logo, .svg to skia
+    "gooroom_product_1x.icon",
+    "gooroom_product.icon",
   ]
 }
 
diff --git a/components/omnibox/browser/vector_icons/gooroom_product.icon b/components/omnibox/browser/vector_icons/gooroom_product.icon
new file mode 100644
index 000000000000..b62188f9d43f
--- /dev/null
+++ b/components/omnibox/browser/vector_icons/gooroom_product.icon
@@ -0,0 +1,47 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+CANVAS_DIMENSIONS, 16,
+MOVE_TO, 7.85f, 15.79f,
+R_ARC_TO, 6.67f, 6.67f, 0, 0, 1, -0.78f, -0.31f,
+R_CUBIC_TO, -0.54f, -0.27f, -1.18f, -0.72f, -1.74f, -1.23f,
+R_ARC_TO, 18.02f, 18.02f, 0, 0, 1, -1.01f, -1.02f,
+CUBIC_TO, 3.19f, 11.96f, 2.27f, 10.35f, 1.8f, 8.84f,
+R_CUBIC_TO, -0.29f, -0.92f, -0.36f, -1.48f, -0.34f, -2.55f,
+R_CUBIC_TO, 0.01f, -0.64f, 0.03f, -0.77f, 0.13f, -1.17f,
+R_CUBIC_TO, 0.38f, -1.48f, 1.52f, -2.96f, 2.99f, -3.88f,
+R_CUBIC_TO, 1.93f, -1.21f, 4.26f, -1.38f, 6.33f, -0.48f,
+R_CUBIC_TO, 1.55f, 0.68f, 2.89f, 1.98f, 3.43f, 3.35f,
+R_CUBIC_TO, 0.09f, 0.22f, 0.15f, 0.48f, 0.18f, 0.68f,
+R_CUBIC_TO, 0.02f, 0.15f, 0.01f, 0.17f, -0.03f, 0.13f,
+R_ARC_TO, 4.55f, 4.55f, 0, 0, 0, -0.34f, -0.18f,
+R_CUBIC_TO, -1.19f, -0.59f, -2.91f, -1.18f, -4.37f, -1.5f,
+R_CUBIC_TO, -1.3f, -0.28f, -2.5f, -0.36f, -3.37f, -0.2f,
+R_CUBIC_TO, -0.59f, 0.1f, -1.13f, 0.34f, -1.44f, 0.64f,
+R_LINE_TO, -0.15f, 0.15f,
+R_LINE_TO, -0.07f, 0.3f,
+R_CUBIC_TO, -0.32f, 1.35f, -0.4f, 2.73f, -0.23f, 3.91f,
+R_CUBIC_TO, 0.07f, 0.47f, 0.14f, 0.7f, 0.33f, 1.13f,
+R_CUBIC_TO, 0.39f, 0.86f, 0.9f, 1.62f, 1.56f, 2.32f,
+R_CUBIC_TO, 0.58f, 0.61f, 1.06f, 1.01f, 1.57f, 1.3f,
+R_CUBIC_TO, 0.2f, 0.11f, 0.2f, 0.11f, 0.28f, 0.08f,
+R_CUBIC_TO, 0.47f, -0.17f, 0.9f, -0.47f, 1.45f, -1.02f,
+R_ARC_TO, 6.94f, 6.94f, 0, 0, 0, 1.5f, -2.22f,
+R_CUBIC_TO, 0.1f, -0.24f, 0.22f, -0.59f, 0.22f, -0.65f,
+R_CUBIC_TO, 0, -0.06f, -0.34f, -0.32f, -0.58f, -0.45f,
+R_CUBIC_TO, -0.54f, -0.28f, -1.12f, -0.39f, -1.94f, -0.39f,
+R_CUBIC_TO, -0.24f, 0, -0.49f, 0.01f, -0.54f, 0.02f,
+R_LINE_TO, -0.1f, 0.02f,
+R_LINE_TO, -0.01f, -0.43f,
+R_CUBIC_TO, -0.01f, -0.24f, -0.02f, -0.69f, -0.02f, -1.01f,
+LINE_TO, 8.2f, 6.13f,
+R_LINE_TO, 0.32f, -0.01f,
+R_CUBIC_TO, 0.5f, -0.02f, 1.23f, 0.04f, 1.84f, 0.16f,
+R_ARC_TO, 9.73f, 9.73f, 0, 0, 1, 3.77f, 1.64f,
+R_CUBIC_TO, 0.45f, 0.33f, 0.41f, 0.27f, 0.38f, 0.45f,
+R_CUBIC_TO, -0.11f, 0.6f, -0.4f, 1.58f, -0.66f, 2.18f,
+R_CUBIC_TO, -0.81f, 1.91f, -2.16f, 3.42f, -4.05f, 4.55f,
+R_CUBIC_TO, -0.42f, 0.25f, -1.19f, 0.63f, -1.36f, 0.68f,
+R_ARC_TO, 1.63f, 1.63f, 0, 0, 1, -0.58f, 0.01f,
+CLOSE
diff --git a/components/omnibox/browser/vector_icons/gooroom_product_1x.icon b/components/omnibox/browser/vector_icons/gooroom_product_1x.icon
new file mode 100644
index 000000000000..b62188f9d43f
--- /dev/null
+++ b/components/omnibox/browser/vector_icons/gooroom_product_1x.icon
@@ -0,0 +1,47 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+CANVAS_DIMENSIONS, 16,
+MOVE_TO, 7.85f, 15.79f,
+R_ARC_TO, 6.67f, 6.67f, 0, 0, 1, -0.78f, -0.31f,
+R_CUBIC_TO, -0.54f, -0.27f, -1.18f, -0.72f, -1.74f, -1.23f,
+R_ARC_TO, 18.02f, 18.02f, 0, 0, 1, -1.01f, -1.02f,
+CUBIC_TO, 3.19f, 11.96f, 2.27f, 10.35f, 1.8f, 8.84f,
+R_CUBIC_TO, -0.29f, -0.92f, -0.36f, -1.48f, -0.34f, -2.55f,
+R_CUBIC_TO, 0.01f, -0.64f, 0.03f, -0.77f, 0.13f, -1.17f,
+R_CUBIC_TO, 0.38f, -1.48f, 1.52f, -2.96f, 2.99f, -3.88f,
+R_CUBIC_TO, 1.93f, -1.21f, 4.26f, -1.38f, 6.33f, -0.48f,
+R_CUBIC_TO, 1.55f, 0.68f, 2.89f, 1.98f, 3.43f, 3.35f,
+R_CUBIC_TO, 0.09f, 0.22f, 0.15f, 0.48f, 0.18f, 0.68f,
+R_CUBIC_TO, 0.02f, 0.15f, 0.01f, 0.17f, -0.03f, 0.13f,
+R_ARC_TO, 4.55f, 4.55f, 0, 0, 0, -0.34f, -0.18f,
+R_CUBIC_TO, -1.19f, -0.59f, -2.91f, -1.18f, -4.37f, -1.5f,
+R_CUBIC_TO, -1.3f, -0.28f, -2.5f, -0.36f, -3.37f, -0.2f,
+R_CUBIC_TO, -0.59f, 0.1f, -1.13f, 0.34f, -1.44f, 0.64f,
+R_LINE_TO, -0.15f, 0.15f,
+R_LINE_TO, -0.07f, 0.3f,
+R_CUBIC_TO, -0.32f, 1.35f, -0.4f, 2.73f, -0.23f, 3.91f,
+R_CUBIC_TO, 0.07f, 0.47f, 0.14f, 0.7f, 0.33f, 1.13f,
+R_CUBIC_TO, 0.39f, 0.86f, 0.9f, 1.62f, 1.56f, 2.32f,
+R_CUBIC_TO, 0.58f, 0.61f, 1.06f, 1.01f, 1.57f, 1.3f,
+R_CUBIC_TO, 0.2f, 0.11f, 0.2f, 0.11f, 0.28f, 0.08f,
+R_CUBIC_TO, 0.47f, -0.17f, 0.9f, -0.47f, 1.45f, -1.02f,
+R_ARC_TO, 6.94f, 6.94f, 0, 0, 0, 1.5f, -2.22f,
+R_CUBIC_TO, 0.1f, -0.24f, 0.22f, -0.59f, 0.22f, -0.65f,
+R_CUBIC_TO, 0, -0.06f, -0.34f, -0.32f, -0.58f, -0.45f,
+R_CUBIC_TO, -0.54f, -0.28f, -1.12f, -0.39f, -1.94f, -0.39f,
+R_CUBIC_TO, -0.24f, 0, -0.49f, 0.01f, -0.54f, 0.02f,
+R_LINE_TO, -0.1f, 0.02f,
+R_LINE_TO, -0.01f, -0.43f,
+R_CUBIC_TO, -0.01f, -0.24f, -0.02f, -0.69f, -0.02f, -1.01f,
+LINE_TO, 8.2f, 6.13f,
+R_LINE_TO, 0.32f, -0.01f,
+R_CUBIC_TO, 0.5f, -0.02f, 1.23f, 0.04f, 1.84f, 0.16f,
+R_ARC_TO, 9.73f, 9.73f, 0, 0, 1, 3.77f, 1.64f,
+R_CUBIC_TO, 0.45f, 0.33f, 0.41f, 0.27f, 0.38f, 0.45f,
+R_CUBIC_TO, -0.11f, 0.6f, -0.4f, 1.58f, -0.66f, 2.18f,
+R_CUBIC_TO, -0.81f, 1.91f, -2.16f, 3.42f, -4.05f, 4.55f,
+R_CUBIC_TO, -0.42f, 0.25f, -1.19f, 0.63f, -1.36f, 0.68f,
+R_ARC_TO, 1.63f, 1.63f, 0, 0, 1, -0.58f, 0.01f,
+CLOSE
diff --git a/components/strings/components_chromium_strings_ko.xtb b/components/strings/components_chromium_strings_ko.xtb
index d30ef37a853e..808ed89b1306 100644
--- a/components/strings/components_chromium_strings_ko.xtb
+++ b/components/strings/components_chromium_strings_ko.xtb
@@ -4,7 +4,7 @@
 <translation id="130631256467250065">기기를 다시 시작할 때 변경사항이 적용됩니다.</translation>
 <translation id="1838412507805038478">Chromium이 <ph name="ISSUER" />에서 이 웹사이트의 인증서를 발행했음을 확인했습니다.</translation>
 <translation id="275588974610408078">비정상 종료 보고서는 Chromium에서 사용할 수 없습니다.</translation>
-<translation id="3064346599913645280">안전한 Chromium 페이지를 보는 중</translation>
+<translation id="32013959327000922">안전한 Gooroom 페이지를 보는 중</translation>
 <translation id="3550966579244642892">Chromium OS 초기 설정이 완료되지 않았습니다.</translation>
 <translation id="4365115785552740256">Chromium은 <ph name="BEGIN_LINK_CHROMIUM" />Chromium<ph name="END_LINK_CHROMIUM" /> 오픈소스 프로젝트를 비롯한 여러 <ph name="BEGIN_LINK_OSS" />오픈소스 소프트웨어<ph name="END_LINK_OSS" />에 기초해 만들어진 브라우저입니다.</translation>
 <translation id="4559775032954821361">Chromium 메뉴 &gt;
@@ -36,4 +36,4 @@
 <translation id="8187289872471304532">애플리케이션 &gt; 시스템 환경설정 &gt; 네트워크 &gt; 고급 &gt; 프록시로 이동한 다음
           선택된 프록시를 선택 취소합니다.</translation>
 <translation id="8684913864886094367">Chromium이 제대로 종료되지 않았습니다.</translation>
-</translationbundle>
\ No newline at end of file
+</translationbundle>
-- 
2.20.1

